src/lsdb/loaders/hipscat/read_hipscat.py

```diff
@@ -66,6 +73,79 @@ def read_hipscat(
     return loader.load_catalog()
 
 
+def read_hipscat_subset(
+    path: str,
+    catalog_type: Type[Dataset] | None = None,
+    storage_options: dict | None = None,
+    search_filter: AbstractSearch | None = None,
+    order: int | None = None,
+    n_pixels: int | None = None,
+    **kwargs,
+) -> Dataset:
+    """Load a catalog subset from a HiPSCat formatted catalog.
+
+    To be used with a filter or (order, n_files), but not both.
+
+    Typical usage example, where we load a catalog from a cone search:
+        lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            filter=lsdb.core.search.ConeSearch(ra, dec, radius_arcsec),
+        )
+
+    Typical usage example, where we load a catalog from a pre-defined number of files:
+         lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            order=5,
+            n_pixels=10,
```

Comment:
This search syntax is not intuitive to me. I think it's because `Npix` is a very specific thing in HiPS. it's also bizarre to me to get those partitions returned that are at a specific order.

alternative a. rename to `num_partitions`?
alternative b. provide some kind of `partition_list`?
alternative c. remove this style of filtering for this PR so we can create a separate discussion about it, because i have even more alternatives and i don't want to de-rail the foundational work.

Reply:
I went with alternative c. to make this PR focus solely on the spatial filtering!